Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/TestsController.cs

```diff
@@ -638,16 +638,11 @@ public ActionResult Import(string problemId, HttpPostedFileBase file, bool retes
 
             if (deleteOldFiles)
             {
-                var tests = problem.Tests.Select(t => new { t.Id, TestRuns = t.TestRuns.Select(tr => tr.Id) }).ToList();
-                foreach (var test in tests)
+                var testsIds = problem.Tests.Select(t => t.Id).ToList();
+                foreach (var testId in testsIds)
                 {
-                    var testRuns = test.TestRuns.ToList();
-                    foreach (var testRun in testRuns)
-                    {
-                        this.Data.TestRuns.Delete(testRun);
-                    }
-
-                    this.Data.Tests.Delete(test.Id);
+                    this.Data.TestRuns.All().Where(tr => tr.TestId == testId).Delete();
```

Comment:
Why don't you use this: https://github.com/SoftUni-Internal/OpenJudgeSystem/blob/master/Open%20Judge%20System/Data/OJS.Data.Contracts/IRepository.cs#L28 ?